File: subit64-ts/encoder.ts
export function encode(axes: Record<string, number>): number {
  const bits = [
    axes.A ?? 0,
    axes.B ?? 0,
    axes.C ?? 0,
    axes.D ?? 0,
    axes.E ?? 0,
    axes.F ?? 0
  ];
  return bits.reduce((acc, bit, i) => acc + (bit << i), 0);
}

File: subit64/encoder.py
def encode(axes: Dict[str, int]) -> int:
    bits = [
        axes.get("A", 0),
        axes.get("B", 0),
        axes.get("C", 0),
        axes.get("D", 0),
        axes.get("E", 0),
        axes.get("F", 0),
    ]
    return sum(bit << i for i, bit in enumerate(bits))
